README.md

```diff
@@ -1,3 +1,138 @@
+
+# How to use SDK
+
+## Initializing SDK
+`VerifiableCredentialSDK` - this class is used to initialize the SDK. You may want to call during your dependency injection initialization:
+```swift
+VerifiableCredentialSDK.initialize();
+```
+
+We currently support the following services:
+
+```swift
+IssuanceService()
+PresentationService()
+```
+
+## APIs
+
+Our APIs use PromiseKit. Read more about promises [here](https://github.com/mxcl/PromiseKit). 
+
+All our public APIs return `Promise<T>` objects. This forces explicit error handling.
+
+You can unpack and handle these promises easily in Swift with the `done` and `catch` statements
+
+```swift
+functionThatReturnPromise().done { objectReturned in
+    handleRequestSuccess(objectReturned) // will be smartcasted into <T>
+}.catch { error in
+    handleRequestFailure(error)
+}
+```
+
+## Receive a Verifiable Credential (IssuanceService)
+
+To receive a verifiable credential you need a service endpoint providing an issuance contract. You can either get it from someone or create your own. See [How to customize your credentials](https://docs.microsoft.com/en-us/azure/active-directory/verifiable-credentials/credential-design) for more information or use an existing provider. In the future, we plan to support the DIF standard [Credential Manifest](https://identity.foundation/credential-manifest/) from the DIF.
+
+```swift
+    func issuanceSample() {
+        /// set up issuance service through dependency injection if you like.
+        let issuanceService = IssuanceService()
+        
+        issuanceService.getRequest(usingUrl: "<issuance request url>").done { issuanceRequest in
+            self.handle(successfulRequest: issuanceRequest, with: issuanceService)
+        }.catch { error in
+            self.handle(failedRequest: error)
+        }
+    }
+    
+    private func handle(successfulRequest request: IssuanceRequest, with service: IssuanceService) {
+        
+        let response: IssuanceResponseContainer
+        
+        do {
+            response = try IssuanceResponseContainer(from: request.content, contractUri: "<issuance request url>")
+        } catch {
+            VCSDKLog.sharedInstance.logError(message: "Unable to create IssuanceResponseContainer.")
+            return
+        }
+        
+        service.send(response: response).done { verifiableCredential in
+            self.handle(successfulResponse: verifiableCredential)
+        }.catch { error in
+            self.handle(failedResponse: error)
+        }
+    }
+```
+
+Most issuance requests will ask you for attestations that the user might need to provide. Provide them by filling the three available maps for self attested claims, idtokens and vcs.
+
+```swift
+private func addRequestedData(response: IssuanceResponseContainer) {
+    response.requestedSelfAttestedClaimMap["key"] = "user specified values"

```

Comment:
nit, Would this add a new value "key" into the map? I think in real code we end up accessing the keys directly as you did here. In my sample I tried to emphasize the idea that you are supposed to reuse the existing keys and just provide the values.

Reply:
I changed the wording on Android slightly to reflect that "Most issuance requests will ask you for attestations that the user might need to provide. Provide them by filling the values for the existing keys in the three available maps for self attested claims, idtokens and vcs."